homework_4_numpy/numpy_challenge.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+# Create three of my favorite arrays.
+if __name__ == "__main__":
+    favorite_arr_1 = np.arange(5, 21, 2)
+    favorite_arr_2 = np.random.randint(20, 35, size=(3, 3))
+    favorite_arr_3 = (np.arange(1, 16)).reshape((3, 5))
+
+# Matrix multiplication
+def matrix_multiplication(matrix_1, matrix_2):
+    return np.matmul(matrix_1, matrix_2)
+
+# Matrix multiplication check
+def multiplication_check(list_of_matrix):
+    for i in range(len(list_of_matrix) - 1):
+        if list_of_matrix[i].shape[1] == list_of_matrix[i+1].shape[0]:
+            possibility = True
+        else:
+            possibility = False
+    return possibility

```

Comment:
В случае хотя бы одного несовпадения размеров, перемножить весь список будет уже невозможно. В твоём же коде может получиться ситуация `True True False True True`, возьмётся последнее значение `possibility`, что будет неверно.
```suggestion
    for i in range(len(list_of_matrix) - 1):
        if list_of_matrix[i].shape[1] != list_of_matrix[i+1].shape[0]:
            return False
    return True

```